add-chain/config/config.go

```diff
@@ -34,14 +34,29 @@ func ConstructChainConfig(
 	if err != nil {
 		return superchain.ChainConfig{}, fmt.Errorf("error reading file: %w", err)
 	}
+
+	// Detect any legacy "plasma" config fields and alert user to adjust their config to match
+	// the new "altda" format
+	var plasmaConfig LegacyPlasma
+	if err = json.Unmarshal(file, &plasmaConfig); err != nil {
+		return superchain.ChainConfig{}, fmt.Errorf("error unmarshaling legacy plasma json: %w", err)
+	}
+	if err = plasmaConfig.CheckNonNilFields(); err != nil {
+		fmt.Printf(
+			"❌ Detected legacy plasma configuration fields in rollup json file.\n" +
+				"❌   - Please follow the guide at the following url to migrate to the new altda format:\n" +
+				"❌   - https://docs.optimism.io/builders/chain-operators/features/alt-da-mode#breaking-changes-renaming-plasma-mode-to-alt-da-mode\n")
+		return superchain.ChainConfig{}, err
+	}
+
```

Comment:
Nit/Optional: we could extract this into a helper and call it from the test?

Reply:
What do you mean by "call it from the test"?

Reply:
Ahh ok I see. I'll keep that in mind for a potential refactor in a follow-up